### PR TITLE
`azurerm_kusto_cluster` - supports double_encryption_enabled, systemAssignedIdentity and systemAssignedUserAssignedIdentity

### PR DESCRIPTION
**Pull request by njuCZ**
support new fields: "double_encryption_enabled"
enhancement for `identity` block: support systemAssignedIdentity and systemAssignedUserAssignedIdentity
once more validation item for sku name

the acctest "TestAccKustoCluster_zones" failed either in master branch, I have emailed the service team

acording to https://docs.microsoft.com/en-us/azure/data-explorer/double-encryption?tabs=portal
> Enabling double encryption is only possible during cluster creation.
> Once infrastructure encryption is enabled on your cluster, you can't disable it.

We could infer that it's a forcenew field

### CODE REVIEW

**Comment by njuCZ** [comment]
the website-test CI has been removed. It seems reopen the PR could not trigger the CI

File: azurerm/internal/services/kusto/identity.go [hunk 1]
```diff
@@ -22,44 +24,63 @@ func schemaIdentity() *schema.Schema {
...
+						string(kusto.IdentityTypeUserAssigned),
+						string(kusto.IdentityTypeSystemAssignedUserAssigned),
 					}, true),
 					DiffSuppressFunc: suppress.CaseDifference,
 				},
+
+				"identity_ids": {
+					Type:     schema.TypeSet,
+					Optional: true,
+					Elem: &schema.Schema{
```
Comment (katbyte) [inline azurerm/internal/services/kusto/identity.go:37]: should this still not also be computed?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: azurerm/internal/services/kusto/identity.go (hunk 1 above)
Comment (njuCZ) [reply azurerm/internal/services/kusto/identity.go:37]: @katbyte this is not a computed field. when the type is `SystemAssigned, UserAssigned`, users could set this property to pass the user managed identity. In this PR, I also add such an example

an example in VM: https://github.com/terraform-providers/terraform-provider-azurerm/blob/2598a62911c599278f06cf3fe9b57224e6bd3339/azurerm/internal/services/compute/virtual_machine.go#L70

Comment (katbyte) [reply azurerm/internal/services/kusto/identity.go:37]: 👍

**Review by katbyte — APPROVED** [review]
Thanks @njuCZ - LGTM 👍

**Pull request merged**